dbaas/dbaas-aggregator/src/main/java/com/netcracker/cloud/dbaas/controller/v3/AggregatedDatabaseAdministrationControllerV3.java

```diff
@@ -396,9 +406,23 @@ public Response deleteDatabaseByClassifier(@Parameter(description = "A unique id
     }
 
     private void checkOriginService(UserRolesServices rolesServices) {
+        if (securityContext.getUserPrincipal() instanceof DefaultJWTCallerPrincipal principal) {
```

Comment:
Add null check for securityContext.getUserPrincipal() to prevent potential NPE.

```suggestion
if (securityContext.getUserPrincipal() != null && securityContext.getUserPrincipal() instanceof DefaultJWTCallerPrincipal principal) {
```

#deepseek-review:inline